### PR TITLE
Post Search and Image Upload

### PR DESCRIPTION
**Pull request by frinzekt**
**Requires:**
**- Creation of text index in the mongoDB configuration**
` db.post.createIndex({title:"text",overview:"text",content:"text", categories:"text"})`

**Feature:**
- Searching of Post by title, content, description, categories
- Base64 Upload (not the best, but it is the easiest)

**Minor Changes/Fixes:**
- Changing of some UI of post
- Undefined Post when Refreshed as Router params is empty or not processed. In simpler terms, when refreshing, the API call is done before the router params variables are loaded resulting in API call with undefined ID
     - Workaround used: forcing the event loop to handle the API call after some events happened
`setTimeOut( [API Call], 0)` (weird, but it works)

### CODE REVIEW

**Comment by frinzekt** [comment]
I am having trouble with `Port 4000 is already in use` because in there is:
- `server.listen(port)` in `./bin/www`
- `app.listen(port)` in `app.js`

I believe I've raised this issue in the other pull request so I am commenting one of them (`server.listen(port)`).

File: backend/app.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ const postRouter = require('./routes/post')
 const app = express()
 
 app.use(logger('dev'))
-app.use(express.json())
+app.use(express.json({ limit: '50mb' }))
```
Comment (noobling) [inline backend/app.js]: might want to bump that down 50 mb is a pretty large file

File: backend/controller/post.js [hunk 2]
```diff
@@ -42,36 +42,45 @@ const getPost = payload => {
 }
 
 //GET A GROUP OF POSTS IN A SPECIFIC PAGE IN THE PAGINATION
-const getPagePost = (page, numberOfPost) => {
+const getPagePost = (page, numberOfPost, searchFilter) => {
+  console.log(searchFilter)
```
Comment (noobling) [inline backend/controller/post.js]: remove ocnsole logs

File: backend/routes/post.js [hunk 3]
```diff
@@ -1,11 +1,11 @@
-var express = require('express')
-var router = express.Router()
+let express = require('express')
```
Comment (noobling) [inline backend/routes/post.js]: If you really one to be a cool kid who writes functional js most your `let` should be `const`

Comment (noobling) [reply backend/routes/post.js]: Normally I would avoid changing `let` to `var` because it is a lot of line changes that could cause nastly merge conflicts

File: frontend/pages/post/[id].js [hunk 4]
```diff
@@ -302,3 +326,7 @@ class LongPost extends React.Component {
 }
 
 export default withRouter(LongPost)
+
+// if (typeof window !== 'undefined') {
+//   import
+// }
```
Comment (noobling) [inline frontend/pages/post/[id].js]: you should try to delete code rather than comment it out

File: frontend/pages/posts/[pageNumber].js [hunk 5]
```diff
@@ -13,21 +12,26 @@ class Posts extends React.Component {
...
     loaded: false,
-    currentCollection: this.props.router.query.pageNumber
+    searchFilter: ''
   }
 
   componentDidMount() {
-    this.updatePostsDisplay()
+    //Forcing Asynchronous Order To Be Executed Last So Router Parameter is not Undefined
+    setTimeout(
+      () => this.updatePostsDisplay(this.props.router.query.pageNumber),
```
Comment (noobling) [inline frontend/pages/posts/[pageNumber].js]: This is a bit of a hack which I don't think would be an issue if you used `functional` components
